banzai_floyds/dbs.py

```diff
@@ -73,3 +86,90 @@ def ingest_standards(db_address):
                           'ra': standard_hdu[0].header['RA'],
                           'dec': standard_hdu[0].header['DEC']}
             add_or_update_record(db_session, FluxStandard, attributes, attributes)
+
+
+def get_order_location(dateobs, order_id, instrument, db_address):
+    with get_session(db_address) as db_session:
+        location_query = db_session.query(OrderLocation).filter(OrderLocation.instrument_id == instrument.id)
+        location_query = location_query.filter(OrderLocation.order_id == order_id)
+        location_query = location_query.filter(OrderLocation.good_after <= dateobs)
+        location_query = location_query.filter(OrderLocation.good_until >= dateobs)
+        location_query.order_by(desc(OrderLocation.id))
+        order_location = location_query.first()
+        order_location = [order_location.xdomainmin, order_location.xdomainmax]
+    return order_location
+
+
+def add_order_location(db_address, instrument_id, xdomainmin, xdomainmax,
+                       order_id, good_after, good_until):
+    """ Add the x range (location) to use for a given order/instrument.
+
+    We cover 4 cases:
+    - Replace the current running location (good until = inf):
+        - The new location will become the current running location
+        - The old location will retain its good after date and set its good until to the good after of the new location
+    - The new range dates fall within a an exisiting range (good until != inf):
+        - We split the exisiting location, one with good after being the original good after and good until being the
```

Comment:
Typos for example "within a an ..." and incorrect spelling of existing a few times in this doc string.